src/main/java/cubicchunks/world/IOpacityIndex.java

```diff
@@ -47,6 +47,16 @@
 	void onOpacityChange(int localX, int blockY, int localZ, int opacity);
 
 	/**
+	 * Returns true if the block at the given position is occluded by a known non-opaque block further up.
+	 *
+	 * @param localX local block x-coordinate (0..15)
+	 * @param blockY global block y-coordinate
+	 * @param localZ local block z-coordinate (0..15)
+	 * @return true if there exists a known non-opaque block above the given y-coordinate
+	 */
+	boolean isOccluded(int localX, int blockY, int localZ);
```

Comment:
Determining if a given block-position is occluded is contextually the OpacityIndex's job. If a different kind of LightManager happens to use the OpacityIndex it'd likely have to implement this method if it weren't already there.